functorch/functorch/_src/aot_autograd.py

```diff
@@ -256,6 +281,12 @@ def fake_fn(primals, tangents):
                     compiled_bw = bw_compiler(bw_module, bw_args)
             else:
                 fw_outs = normalize_as_list(compiled_fw(*flat_tensor_args))
+                if config.use_functionalize:
+                    # Call the input mutation submodule with the mutated inputs
+                    mutated_inputs = fw_outs[:num_mutated_inputs]
+                    fw_outs = fw_outs[num_mutated_inputs:]
+                    mutated_input_args = [x for pair in zip(original_inputs_needing_mutation, mutated_inputs) for x in pair]
+                    input_mutation_submodule(*mutated_input_args)
```

Comment:
It's lucky that AOTAutograd is the only place using this, because the calling convention is pretty complicated and I definitely would not be wanting to force arbitrarily many clients to have to deal with this by hand. I guess this was the appeal of putting it in codegen: if you did this in codegen, the return signature of the module doesn't actually change and all of the extra bookkeeping is hidden from the user (at the cost of needing to know that you have to preserve the codegen attribute across passes, and at the cost of needing to know that if you want to retrace without seeing the `copy_` calls, you have to actually run the interpreter. This second one is not such a big deal, since @SherlockNoMad is about to require you to do that anyway--see my NACK on https://github.com/pytorch/pytorch/pull/82755 )